amazonka/src/MRIO/Amazonka.hs

```diff
@@ -1,28 +1,73 @@
+{-# OPTIONS -Wno-orphans #-}
+
 module MRIO.Amazonka
-  ( HasAWS(..)
+  ( AWS.Env
+  , HasAWS
+  , HasAWSEnv(..)
+  , HasResourceMap(..)
+  , ResourceMap
   , paginate
+  , runAWS
   , send
+  , withResourceMap
   )
 where
 
-import           Data.Conduit                   ( ConduitT
-                                                , transPipe
-                                                )
-import           Data.Function                  ( ($) )
+import           Control.Monad                         ( (=<<) )
+import           Control.Monad.IO.Unlift               ( MonadUnliftIO
+                                                       , liftIO
+                                                       , withRunInIO
+                                                       )
+import           Control.Monad.Reader                  ( asks )
+import           Control.Monad.Trans.Resource          ( runResourceT )
+import           Control.Monad.Trans.Resource.Internal ( MonadResource
+                                                       , ReleaseMap
+                                                       , ResourceT(..)
+                                                       , liftResourceT
+                                                       )
+import           Data.Conduit                          ( ConduitT
+                                                       , transPipe
+                                                       )
+import           Data.Function                         ( ($), (.) )
+import           Data.IORef                            ( IORef )
 import           MRIO.Core
-import           Network.AWS                    ( AWS
-                                                , AWSPager
-                                                , AWSRequest
-                                                , Rs
-                                                )
 
-import qualified Network.AWS                   as AWS
+import qualified Network.AWS as AWS
+
+class HasAWSEnv env where
+  awsEnv :: env -> AWS.Env
+
+type ResourceMap = IORef ReleaseMap
+
+class HasResourceMap env where
```

Comment:
@epicallan In the future we may want to split this into an `mrio-resourcet` package.